File: examples/scripts/orca/participant.py
import numpy as np
from mathutils import normalize


class Participant(object):
    def __init__(self, position: np.ndarray, velocity: np.ndarray, radius: np.float64, vmax=None, target=None) -> None:
        super().__init__()
        self.position = position
        self.velocity = velocity
        self.radius = radius
        self.vmax = vmax
        self.target = target

    def update_position(self, position: np.ndarray) -> None:
        self.position = position
        self.velocity = self.target - self.position
        if np.linalg.norm(self.velocity) > self.vmax:
            self.velocity = normalize(self.velocity) * self.vmax

File: examples/scripts/rust-orca/src/participant.rs
use crate::math;

pub struct Participant {
    pub position: ndarray::Array1<f64>,
    pub velocity: ndarray::Array1<f64>,
    pub radius: f64,
    pub safezone: f64,
    pub vmax: f64,
    pub target: ndarray::Array1<f64>,
}

impl Participant {
    pub fn update_position(&mut self, position: &ndarray::Array1<f64>) {
        self.position = position.clone();
        self.velocity = &self.target - &self.position;
        if math::norm(&self.velocity) > self.vmax {
            self.velocity = math::normalize(&self.velocity) * self.vmax
        }
    }
}
